pytensor/link/pytorch/dispatch/subtensor.py

```diff
@@ -0,0 +1,38 @@
+from pytensor.link.pytorch.dispatch.basic import pytorch_funcify
+from pytensor.tensor.subtensor import (
+    AdvancedSubtensor,
+    AdvancedSubtensor1,
+    Subtensor,
+    indices_from_subtensor,
+)
+
+
+@pytorch_funcify.register(Subtensor)
+@pytorch_funcify.register(AdvancedSubtensor)
+@pytorch_funcify.register(AdvancedSubtensor1)
+def pytorch_funcify_Subtensor(op, node, **kwargs):
+    idx_list = getattr(op, "idx_list", None)
+
+    def subtensor(x, *ilists):
+        indices = indices_from_subtensor(ilists, idx_list)
+        new_indices = []
+
+        for i in indices:
+            new_start, new_step, new_stop = None, None, None
+            if isinstance(i, slice):
+                if i.start:
+                    new_start = i.start.item()
+                if i.step:
+                    new_step = i.step.item()
+                if i.stop:
+                    new_stop = i.stop.item()
+                new_indices.append(slice(new_start, new_stop, new_step))
```

Comment:
Why is this needed? I think the slices should be scalars already?

Reply:
`start`, `step`, `stop` are pytorch tensors (example: `slice(tensor(1), tensor(2), None)`) and have to be converted to scalars